cmake/bob_robotics.cmake

```diff
@@ -80,77 +80,88 @@ macro(BoB_project)
 
     # If this project includes a GeNN model...
     if(PARSED_ARGS_GENN_MODEL OR PARSED_ARGS_INCLUDE_GENN_USERPROJECTS)
-        # Find genn-buildmodel (which should be in the path)
         if(WIN32)
-            find_program(GENN_BUILDMODEL genn-buildmodel.bat)
+            set(shext bat)
         else()
-            find_program(GENN_BUILDMODEL genn-buildmodel.sh)
+            set(shext sh)
         endif()
-        if(GENN_BUILDMODEL)
-            # Remove filename to get path to GeNN bin directory
-            get_filename_component(GENN_BIN_PATH ${GENN_BUILDMODEL} DIRECTORY)
-
-            # Get absolute path to userproject include
-            get_filename_component(GENN_USERPROJECT ${GENN_BIN_PATH}/../userproject/include ABSOLUTE)
-            message("GeNN found in ${GENN_USERPROJECT}")
-            BoB_add_include_directories(${GENN_USERPROJECT})
-            # If th
-            if(PARSED_ARGS_INCLUDE_GENN_USERPROJECTS)
-                # On *nix link dl
-                if(NOT WIN32)
-                    BoB_add_link_libraries(dl)
-                endif()
-            elseif(PARSED_ARGS_GENN_MODEL)
-                get_filename_component(genn_model_name "${CMAKE_CURRENT_SOURCE_DIR}" NAME)
-                set(genn_model_dir "${CMAKE_CURRENT_BINARY_DIR}/${genn_model_name}_CODE")
-                set(genn_model_src "${CMAKE_CURRENT_SOURCE_DIR}/${PARSED_ARGS_GENN_MODEL}")
-                set(genn_model_dest "${genn_model_dir}/runner.cc")
-
-                if(NOT GENN_CPU_ONLY)
-                    if(DEFINED ENV{CPU_ONLY} AND NOT $ENV{CPU_ONLY} STREQUAL 0)
-                        set(GENN_CPU_ONLY TRUE)
-                    else()
-                        set(GENN_CPU_ONLY ${PARSED_ARGS_GENN_CPU_ONLY})
-                    endif()
-                endif(NOT GENN_CPU_ONLY)
-                if(GENN_CPU_ONLY)
-                    message("Building GeNN model for CPU only")
-                    add_definitions(-DCPU_ONLY)
-                    set(CPU_FLAG -c)
+
+        if(DEFINED GENN_PATH)
+            set(GENN_BUILDMODEL "${GENN_PATH}/bin/genn-buildmodel.${shext}")
+        else()
+            # Find genn-buildmodel (which should be in the path)
+            find_program(GENN_BUILDMODEL genn-buildmodel.${shext})
+            if(NOT DEFINED GENN_BUILDMODEL)
+                message(FATAL_ERROR "GeNN not found. Please install and ensure it is in path or set the GENN_PATH CMake variable.")
+            endif()
+        endif()
+
+        # Remove filename to get path to GeNN bin directory
+        get_filename_component(GENN_BIN_PATH ${GENN_BUILDMODEL} DIRECTORY)
+
+        # Get absolute path to userproject include
+        get_filename_component(GENN_USERPROJECT ${GENN_BIN_PATH}/../userproject/include ABSOLUTE)
+        message("GeNN found in ${GENN_USERPROJECT}")
+        BoB_add_include_directories(${GENN_USERPROJECT})
```

Comment:
this could probably go in the ``else`` clause above - if you have ``GENN_PATH``, getting ``GENN_USER_PROJECT`` is trivial 😄

Reply:
Good point!